File: validAnagram.java
/* Given two strings s and t, return true if t is an

of s, and false otherwise.



Example 1:

Input: s = "anagram", t = "nagaram"

Output: true

Example 2:

Input: s = "rat", t = "car"

Output: false
*/
class Solution {
    public boolean isAnagram(String s, String t) {
    //Sorting approach is more straight forward.
    char[] a = s.toCharArray();
    Arrays.sort(a);
    char[] b = t.toCharArray();
    Arrays.sort(b);
    return new String(a) == new String(b);
}

File: validAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()) {
            return false;
        }
        unordered_map<char, int> freq;
        int n = s.length();
        for (int i = 0; i < n; i++) {
            freq[s[i]]++;
        }
        for (int i = 0; i < n; i++) {
            freq[t[i]]--;
        }

        for (int i = 0; i < n; i++) {
            if (freq[s[i]] != 0) {
                return false;
            }
        }
        return true;
    }
};
